### PR TITLE
fix(resource-proxy): Use list options for listing resources

### LINKED ISSUE #766: Events not being filtered to specific resource
**Describe the bug**

The agent is not filtering events for the specific resource being viewed, rather it returns all the events that it has access to. This results in thousands of events when using the agent with cluster-admin rights and potentially impacts performance.

Image

In this example the resource is showing thousands of events but if you get the events for the resource using kubectl you can see there are only two events for the specific resource.

```
$ kubectl events --for=externalsecret/thanos-object-storage
LAST SEEN             TYPE      REASON         OBJECT                                 MESSAGE
21m (x8 over 24m)     Warning   UpdateFailed   ExternalSecret/thanos-object-storage   could not apply template: could not execute template: could not execute template: unable to execute template at key thanos.yaml: unable to parse template at key thanos.yaml: template: thanos.yaml:6: unexpected "\\" in operand
5m51s (x5 over 21m)   Warning   UpdateFailed   ExternalSecret/thanos-object-storage   error processing spec.data[0] (key: acm-observability), err: property access_key does not exist in data of secret "acm-observability"
```

**Steps to reproduce the behaviour**
1. Give the argocd-agent cluster-admin permissions 
2. View the events for a specific resource and note that it is showing all events.
3. (...)

**Expected behavior**
It should only show events for the specific resource.

**Additional context**
This may be what is causing the large messages in #748

### PR DESCRIPTION
**Pull request by jannfis**
**What does this PR do / why we need it**:

The resource proxy on the agent would ignore any query params when listing resources. In the case of Events, this would lead to it returning all events that it had access to (potentially cluster-wide), not the ones that were requested.

**Which issue(s) this PR fixes**:

Fixes #766

**How to test changes / Special notes to the reviewer**:

**Checklist**

* [ ] Documentation update is required by this PR (and has been updated) OR no documentation update is required.

## Summary by CodeRabbit

* **New Features**
  * Resource listing now supports query parameters: fieldSelector, labelSelector, resourceVersion, and pagination (limit, continue) for filtered and paged views.

* **Bug Fixes**
  * Unknown query keys are ignored and emitted as warnings; nil or missing params no longer cause failures.

* **Tests**
  * Added tests for param parsing, propagation of filters/pagination, invalid values handling, and nil/edge cases.

### CODE REVIEW

File: agent/resource.go [hunk 1]
```diff
@@ -331,10 +331,37 @@ func (a *Agent) getManagedResource(ctx context.Context, gvr schema.GroupVersionR
 	return res, err
 }
 
+// listOptionsFromParams builds a v1.ListOptions from the given HTTP query
+// parameter map. Unknown keys are silently ignored.
```
Comment (jgwest) [inline agent/resource.go]: > Unknown keys are silently ignored. 

WDYT about logging at 'warning' level for unknown keys? This would allow us to better detect missing support for any other argocd api resource request calls. (This behaviour is also what I went with for redis proxy for [reasons](https://github.com/argoproj-labs/argocd-agent/issues/751#issuecomment-3926613903) )

**Review by jgwest — CHANGES_REQUESTED** [review]

File: agent/resource.go (hunk 1 above)
Comment (jannfis) [reply agent/resource.go]: Good thinking. I think it cannot hurt to validate the incoming request params to see if something is fishy. Usually, it shouldn't happen that an unknown request parameter is being sent - so warning level makes sense to me.
